### ISSUE TITLE
Verify compliant metric SDK specification implementation: Concurrency requirements

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go#3672]
- [x] Identify all the normative requirements, recommendations, and options [the specification] defines as comments to this issue
- [ ] Ensure the [current metric SDK implementation] is compliant with these normative requirements, recommendations, and options in those comments.

[the specification]: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/metrics/sdk.md
[current metric SDK implementation]: https://pkg.go.dev/go.opentelemetry.io/otel/sdk/metric@v0.39.0

**Comment by pellared** [comment]
> MeterProvider - Meter creation, ForceFlush and Shutdown are safe to be called concurrently.

_TODO_

**Comment by pellared** [comment]
> ExemplarFilter - all methods are safe to be called concurrently.

_TODO_

**Comment by pellared** [comment]
> ExemplarReservoir - all methods are safe to be called concurrently.

> TODO

**Comment by pellared** [comment]
> MetricReader - Collect and Shutdown are safe to be called concurrently.

_TODO_

**Comment by pellared** [comment]
> MetricExporter - ForceFlush and Shutdown are safe to be called concurrently.

_TODO_

**Comment by pellared** [comment]
Currently blocked by:

- [ ] https://github.com/open-telemetry/opentelemetry-go/issues/4246
- [ ] https://github.com/open-telemetry/opentelemetry-go/issues/4248

**Comment by MadVikingGod** [comment]
Exemplars are an experimental feature and not in scope for the current GA.  If Exemplars are the only thing left we should maybe spin those into a different ticket in the Post GA project.

**Comment by pellared** [comment]
Closing as exemplars are not stable.
